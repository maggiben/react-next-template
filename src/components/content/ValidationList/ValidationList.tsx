import styled from "styled-components";
import { 
  Grid, 
  GridItem, 
  Label,
  TextBody,
  AlertIcon,
  CheckIcon
} from '@fravega-it/bumeran-ds-fvg'
import { useTranslation } from 'react-i18next';
import { Person } from 'types/type';
import { SpaceRight, SpaceTop } from "../../Spacing/Spacing";

const Card = styled.div`
  width: 100%;
`;

const Centered = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;

const StatusIcon = (type: Boolean) => type ? <CheckIcon /> : <AlertIcon />;

type ClientCardLayoutProps = {
  person: Person;
}

const ValidationList = (props: ClientCardLayoutProps) => {
  const { t } = useTranslation();
  const { person } = props;
  const validationLayout = {
    email: {
      icon: StatusIcon(person.validations.email),
      color: person.validations.email ? 'green' : 'red' as 'red' | 'green',
      label: person.validations.email ? t('validated email') : t('not validated email'),
      description: person.validations.email ? t('validated email description') : t('not validated email description')
    },
    renaper: {
      icon: StatusIcon(person.validations.renaper),
      color: person.validations.renaper ? 'green' : 'red' as 'red' | 'green',
      label: person.validations.renaper ? t('validated renaper') : t('not validated renaper'),
      description: person.validations.renaper ? t('validated renaper description') : t('not validated renaper description')
    },
    facephi: {
      icon: StatusIcon(person.validations.facephi),
      color: person.validations.facephi ? 'green' : 'red' as 'red' | 'green',
      label: person.validations.facephi ? t('validated facephi') : t('not validated facephi'),
      description: person.validations.facephi ? t('validated facephi description') : t('not validated facephi description')
    }
  };

  return (
    <Card>
      {
        Object.entries(validationLayout).map(([validation, properties], index) => (
          <div key={index}>
            <Grid spacing="s" rowSpacing="s">
              <GridItem xs={12} alignSelf="center" justifySelf="start" >
                <Centered>
                  <Label leftIcon={properties.icon} label={properties.label} color={properties.color}/>
                  <SpaceRight size="xs" />
                  <TextBody size="m" color="neutral" as="span">{properties.description}</TextBody>
                </Centered>
              </GridItem>
            </Grid>
            <SpaceTop size="xs" />
          </div>
        ))
      }
    </Card>
  );
};

  export default ValidationList;
