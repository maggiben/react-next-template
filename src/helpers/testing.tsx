import { default as baseRenderer } from 'react-test-renderer';
import { render, RenderResult } from '@testing-library/react';
import i18n from '@utils/i18n';
import { I18nextProvider } from 'react-i18next';
import { RecoilRoot } from 'recoil';
import { ThemeProvider } from 'styled-components';
import { theme, IDefaultTheme } from '@fravega-it/bumeran-ds-fvg';
import React, { ReactElement } from 'react';
import '@testing-library/jest-dom'

/**
 * Helper used to render components for testing.
 * The function appends the theme to the children.
 * */
interface IRenderer {
  create: (children: ReactElement) => RenderResult;
  baseRenderer: (children: ReactElement) => baseRenderer.ReactTestRenderer;
  renderWithTheme: (children: ReactElement, theme: IDefaultTheme) => RenderResult;
  renderWithI18n: (children: ReactElement) => RenderResult;
}

export const renderer: IRenderer = {
  create: (children: ReactElement): RenderResult =>
    render(
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
  ),
  renderWithTheme: (children: ReactElement, theme: IDefaultTheme): RenderResult =>
    render(
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
  ),
  baseRenderer: (children: ReactElement): baseRenderer.ReactTestRenderer =>
    baseRenderer.create(
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
  ),
  renderWithI18n: (children: ReactElement): RenderResult => 
    render(
      <ThemeProvider theme={theme}>
        <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
      </ThemeProvider>
  ),
};