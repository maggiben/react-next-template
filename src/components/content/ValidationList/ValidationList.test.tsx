import React from 'react';
import { renderer } from '@helpers/testing';
import '@testing-library/jest-dom';
import ValidationList from './ValidationList';
import { expect } from '@jest/globals';

// Mock the utility function used in the component
// jest.mock('@utils/array', () => ({
//   splitIntoTuples: jest.fn(data => data), // Mock the splitIntoTuples function to return the input data
// }));

describe('DataContainer', () => {
  const person = {
    id: '123',
    name: 'John',
    lastname: 'Doe',
    age: '30',
    validations: {
      email: true,
      renaper: false,
      facephi: true,
    },
    identification: {
      type: 'ID',
      number: 123456789,
      expiration: '2022-12-31',
    },
    email: {
      address: 'john.doe@example.com',
      confirmed: true,
      expiration: '2022-12-31',
    },
    selected: false,
    status: {
      label: 'Active',
      color: 'green',
    },
    phone: {
      number: 987654321,
      prefix: 1,
      codeArea: 123,
    },
    address: {
      zipCode: '12345',
      street: 'Main Street',
      number: '123',
      floor: '2',
      apartment: 'A',
      city: 'Cityville',
      state: 'CA',
      country: 'USA',
    },
  };

  it('renders the DataContainer component', () => {
    const tree = renderer.create(<ValidationList person={person} />);
    expect(tree).toMatchSnapshot();
  });

});
