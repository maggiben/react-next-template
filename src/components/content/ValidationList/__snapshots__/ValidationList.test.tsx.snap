// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataContainer renders the DataContainer component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ValidationList__Container-sc-zx6xp2-0 jWPspH"
      >
        <div>
          <div
            class="sc-jsJBEP eNXzoM"
            spacing="s"
          >
            <div
              class="sc-eeDRCY gNxwjx"
            >
              <div
                class="ValidationList__Centered-sc-zx6xp2-1 fOVA-Do"
              >
                <div
                  class="sc-bmzYkS evAJPt"
                  color="green"
                >
                  <svg
                    class="sc-kpDqfm fxbHNO"
                    color="green"
                    fill="none"
                    viewBox="0 0 28 28"
                    width="20px"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M19.996 9.56a.778.778 0 0 1 0 1.1l-7.78 7.78a.778.778 0 0 1-1.1 0l-3.112-3.112a.778.778 0 0 1 1.1-1.1l2.562 2.562 7.23-7.23a.778.778 0 0 1 1.1 0Z"
                      fill-rule="evenodd"
                    />
                  </svg>
                  <span
                    class="sc-iGgWBj djeqlO"
                    color="inherit"
                  >
                    Email Validado
                  </span>
                </div>
                <div
                  class="Spacing__SpaceRight-sc-11df7di-0 bsMTuc"
                />
                <span
                  class="sc-iGgWBj emVxnH"
                  color="neutral"
                >
                  El cliente ya validó su correo electrónico
                </span>
              </div>
            </div>
          </div>
          <div
            class="Spacing__SpaceTop-sc-11df7di-1 gIigKZ"
          />
        </div>
        <div>
          <div
            class="sc-jsJBEP eNXzoM"
            spacing="s"
          >
            <div
              class="sc-eeDRCY gNxwjx"
            >
              <div
                class="ValidationList__Centered-sc-zx6xp2-1 fOVA-Do"
              >
                <div
                  class="sc-bmzYkS hwfFMR"
                  color="red"
                >
                  <svg
                    class="sc-kpDqfm dFcenU"
                    color="red"
                    fill="none"
                    viewBox="0 0 28 28"
                    width="20px"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M11.98 8.168c.898-1.555 3.142-1.555 4.04 0l5.387 9.331c.898 1.555-.224 3.5-2.02 3.5H8.613c-1.796 0-2.918-1.945-2.02-3.5l5.387-9.33Zm2.693.778a.778.778 0 0 0-1.346 0l-5.388 9.33a.778.778 0 0 0 .674 1.167h10.774a.778.778 0 0 0 .674-1.166l-5.388-9.331ZM14 11.667c.43 0 .778.348.778.778V14a.778.778 0 0 1-1.556 0v-1.555c0-.43.349-.778.778-.778Zm-.778 5.443c0-.43.349-.777.778-.777h.008a.778.778 0 0 1 0 1.555H14a.778.778 0 0 1-.778-.778Z"
                      fill-rule="evenodd"
                    />
                  </svg>
                  <span
                    class="sc-iGgWBj djeqlO"
                    color="inherit"
                  >
                    No Validado en Renaper
                  </span>
                </div>
                <div
                  class="Spacing__SpaceRight-sc-11df7di-0 bsMTuc"
                />
                <span
                  class="sc-iGgWBj emVxnH"
                  color="neutral"
                >
                  El cliente no validó su identidad en el Registro Nacional de las Personas
                </span>
              </div>
            </div>
          </div>
          <div
            class="Spacing__SpaceTop-sc-11df7di-1 gIigKZ"
          />
        </div>
        <div>
          <div
            class="sc-jsJBEP eNXzoM"
            spacing="s"
          >
            <div
              class="sc-eeDRCY gNxwjx"
            >
              <div
                class="ValidationList__Centered-sc-zx6xp2-1 fOVA-Do"
              >
                <div
                  class="sc-bmzYkS evAJPt"
                  color="green"
                >
                  <svg
                    class="sc-kpDqfm fxbHNO"
                    color="green"
                    fill="none"
                    viewBox="0 0 28 28"
                    width="20px"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M19.996 9.56a.778.778 0 0 1 0 1.1l-7.78 7.78a.778.778 0 0 1-1.1 0l-3.112-3.112a.778.778 0 0 1 1.1-1.1l2.562 2.562 7.23-7.23a.778.778 0 0 1 1.1 0Z"
                      fill-rule="evenodd"
                    />
                  </svg>
                  <span
                    class="sc-iGgWBj djeqlO"
                    color="inherit"
                  >
                    Validado en Facephi
                  </span>
                </div>
                <div
                  class="Spacing__SpaceRight-sc-11df7di-0 bsMTuc"
                />
                <span
                  class="sc-iGgWBj emVxnH"
                  color="neutral"
                >
                  El cliente ya validó su identidad biométrica
                </span>
              </div>
            </div>
          </div>
          <div
            class="Spacing__SpaceTop-sc-11df7di-1 gIigKZ"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ValidationList__Container-sc-zx6xp2-0 jWPspH"
    >
      <div>
        <div
          class="sc-jsJBEP eNXzoM"
          spacing="s"
        >
          <div
            class="sc-eeDRCY gNxwjx"
          >
            <div
              class="ValidationList__Centered-sc-zx6xp2-1 fOVA-Do"
            >
              <div
                class="sc-bmzYkS evAJPt"
                color="green"
              >
                <svg
                  class="sc-kpDqfm fxbHNO"
                  color="green"
                  fill="none"
                  viewBox="0 0 28 28"
                  width="20px"
                >
                  <path
                    clip-rule="evenodd"
                    d="M19.996 9.56a.778.778 0 0 1 0 1.1l-7.78 7.78a.778.778 0 0 1-1.1 0l-3.112-3.112a.778.778 0 0 1 1.1-1.1l2.562 2.562 7.23-7.23a.778.778 0 0 1 1.1 0Z"
                    fill-rule="evenodd"
                  />
                </svg>
                <span
                  class="sc-iGgWBj djeqlO"
                  color="inherit"
                >
                  Email Validado
                </span>
              </div>
              <div
                class="Spacing__SpaceRight-sc-11df7di-0 bsMTuc"
              />
              <span
                class="sc-iGgWBj emVxnH"
                color="neutral"
              >
                El cliente ya validó su correo electrónico
              </span>
            </div>
          </div>
        </div>
        <div
          class="Spacing__SpaceTop-sc-11df7di-1 gIigKZ"
        />
      </div>
      <div>
        <div
          class="sc-jsJBEP eNXzoM"
          spacing="s"
        >
          <div
            class="sc-eeDRCY gNxwjx"
          >
            <div
              class="ValidationList__Centered-sc-zx6xp2-1 fOVA-Do"
            >
              <div
                class="sc-bmzYkS hwfFMR"
                color="red"
              >
                <svg
                  class="sc-kpDqfm dFcenU"
                  color="red"
                  fill="none"
                  viewBox="0 0 28 28"
                  width="20px"
                >
                  <path
                    clip-rule="evenodd"
                    d="M11.98 8.168c.898-1.555 3.142-1.555 4.04 0l5.387 9.331c.898 1.555-.224 3.5-2.02 3.5H8.613c-1.796 0-2.918-1.945-2.02-3.5l5.387-9.33Zm2.693.778a.778.778 0 0 0-1.346 0l-5.388 9.33a.778.778 0 0 0 .674 1.167h10.774a.778.778 0 0 0 .674-1.166l-5.388-9.331ZM14 11.667c.43 0 .778.348.778.778V14a.778.778 0 0 1-1.556 0v-1.555c0-.43.349-.778.778-.778Zm-.778 5.443c0-.43.349-.777.778-.777h.008a.778.778 0 0 1 0 1.555H14a.778.778 0 0 1-.778-.778Z"
                    fill-rule="evenodd"
                  />
                </svg>
                <span
                  class="sc-iGgWBj djeqlO"
                  color="inherit"
                >
                  No Validado en Renaper
                </span>
              </div>
              <div
                class="Spacing__SpaceRight-sc-11df7di-0 bsMTuc"
              />
              <span
                class="sc-iGgWBj emVxnH"
                color="neutral"
              >
                El cliente no validó su identidad en el Registro Nacional de las Personas
              </span>
            </div>
          </div>
        </div>
        <div
          class="Spacing__SpaceTop-sc-11df7di-1 gIigKZ"
        />
      </div>
      <div>
        <div
          class="sc-jsJBEP eNXzoM"
          spacing="s"
        >
          <div
            class="sc-eeDRCY gNxwjx"
          >
            <div
              class="ValidationList__Centered-sc-zx6xp2-1 fOVA-Do"
            >
              <div
                class="sc-bmzYkS evAJPt"
                color="green"
              >
                <svg
                  class="sc-kpDqfm fxbHNO"
                  color="green"
                  fill="none"
                  viewBox="0 0 28 28"
                  width="20px"
                >
                  <path
                    clip-rule="evenodd"
                    d="M19.996 9.56a.778.778 0 0 1 0 1.1l-7.78 7.78a.778.778 0 0 1-1.1 0l-3.112-3.112a.778.778 0 0 1 1.1-1.1l2.562 2.562 7.23-7.23a.778.778 0 0 1 1.1 0Z"
                    fill-rule="evenodd"
                  />
                </svg>
                <span
                  class="sc-iGgWBj djeqlO"
                  color="inherit"
                >
                  Validado en Facephi
                </span>
              </div>
              <div
                class="Spacing__SpaceRight-sc-11df7di-0 bsMTuc"
              />
              <span
                class="sc-iGgWBj emVxnH"
                color="neutral"
              >
                El cliente ya validó su identidad biométrica
              </span>
            </div>
          </div>
        </div>
        <div
          class="Spacing__SpaceTop-sc-11df7di-1 gIigKZ"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
